continuous-integration/requirements.Rmd

```diff
@@ -0,0 +1,28 @@
+---
+title: "Requirements for continuous integration"
+maintainer: "Thierry Onkelinx"
+output: html_document
+---
+
+# Must have
+
+- start automatically when pushing a git repository on Github or Bitbucket
+- start automatically when creating a pull request on Github or Bitbucket
+- able to use a Docker image to setup the environment in which CI runs
```

Comment:
- AFAIK I'm currently the only one using Docker at INBO.
- Docker is licensed under Apache 2.0 so usage is free. Potential costs are storing private images and running images on AWS. The latter is rather the cost of the AWS infrastructure and not Docker.
- I see the huge advantages for CI and running things in production. There is little to no advantage for ad hoc analysis.
- Docker has limited advantages for researcher. Except that we could use Docker to setup a standard INBO environment on something like AWS.  E.g. RStudio Server is available as a docker image: https://hub.docker.com/r/rocker/rstudio/